packages/client/src/workflow-options.ts

```diff
@@ -39,6 +39,10 @@ export interface WorkflowOptions extends CommonWorkflowOptions {
   followRuns?: boolean;
 }
 
+export interface WorkflowUpdateOptions {
+  readonly id?: string;
```

Comment:
```suggestion
  /** The Update Id, which is used as a per-Workflow Execution idempotency key. If this is not provided, it will be set by the Server, and there is a possiblity that duplicate Updates will be created. */
  readonly id?: string;
```

from talking with spencer, here's my understanding of the id. if it's accurate, it seems like it would be better for the sdk to generate an id clientside when not provided.

Reply:
@lorensr It's already that way ([see here](https://github.com/dandavison/temporalio-sdk-typescript/blob/0db3f0f69e2cc98606ceb63221708fb553034138/packages/client/src/workflow-client.ts#L731)).

Technically, the possibility of duplicate updates remains, though at a higher level; for example:
- a webapp makes a call to some rest api which calls update on some workflow;
- the web service or the SDK generate a random request id;
- the web service request fail at some point after the update got received by the server, so the webapp don't receive a confirmation;
- the client retry the request, which results in a new `update` request to the workflow, with a different request id.

The solution to that is that request id should be generated by the webapp. In some case, it may also be pertinent for the web service to also append some deterministic prefix (eg. a web session id) to avoid security issues that could result from maliciously forged request id.

Same can happen if a Temporal activity is used to send an update to a workflow, and request id is not specified explicitly or is randomly generated by the activity (rather than being deterministically generated by the activity).

Reply:
Update ID is like a combination of workflow ID and request ID, but from a user POV it should be (mostly) seen as like workflow ID and we encourage its use. They should set it to a business identifier. But we do not have a way to fail an update on duplicate ID. There was a lot of discussion server side about update IDs use for addressability and deduplication, and I think what they have is fine.

Reply:
> from talking with spencer, here's my understanding of the id. if it's accurate,

Right, as @mjameswh said, the update ID is set by the client if it's not provided.

Reply:
>  But we do not have a way to fail an update on duplicate ID.

Sounds like it's not an idempotency key? I was going by this server comment:

"An ID with workflow-scoped uniqueness for this update"

https://api-docs.temporal.io/#temporal.api.update.v1.Meta

Reply:
> Sounds like it's not an idempotency key?

It is

> I was going by this server comment:

That comment doesn't say anything about idempotency. To compare with workflows, the update ID is like the workflow ID and the request ID combined into one.

Reply:
> An ID with workflow-scoped uniqueness for this update

I took this to mean that Run-scoped uniqueness is preserved they server, so if I send two updates to the Run with the same ID, the second update will fail. 

> But we do not have a way to fail an update on duplicate ID.

Does this mean the second update will succeed?